plugins/field-multilineinput/test/index.ts

```diff
@@ -9,30 +9,200 @@
+        spellcheck: false,
+        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
+do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
+      },
+    ],
+    output: null,
+    style: 'math_blocks',
+  },
+  {
+    type: 'test_parent_block',
+    message0: 'parent %1',
+    args0: [
+      {
+        type: 'input_value',
+        name: 'INPUT',
+      },
+    ],
+    previousStatement: null,
+    nextStatement: null,
+    style: 'loop_blocks',
+  },
+];
+
+Blockly.defineBlocksWithJsonArray(testBlockDefinitions);
+
+/**
+ * A test toolbox containing the exported blocks and a variety of
+ * test blocks to exercise the multiline text input field in different
+ * contexts (on a shadow block, as the only field on a block, etc).
+ * These are in a simple toolbox, rather than a category toolbox, so that
+ * they are all instantiated every time the test page is opened.
+ */
+const jsonToolbox = {
+  contents: [
+    {
+      kind: 'label',
+      text: 'Exported blocks',
+    },
+    {
+      kind: 'block',
+      type: 'text_multiline',
+    },
+    {
+      kind: 'label',
+      text: 'Test blocks: spellchecked field',
+    },
+    {
+      kind: 'block',
+      type: 'test_spellchecked_field',
+    },
+    {
+      kind: 'block',
+      type: 'test_parent_block',
+      inputs: {
+        INPUT: {
+          shadow: {
+            type: 'test_spellchecked_field',
+          },
+        },
+      },
+    },
+    {
+      kind: 'block',
+      type: 'test_spellchecked_field_and_label',
+    },
+    {
+      kind: 'block',
+      type: 'test_parent_block',
+      inputs: {
+        INPUT: {
+          shadow: {
+            type: 'test_spellchecked_field_and_label',
+          },
+        },
+      },
+    },
+    {
+      kind: 'label',
+      text: 'Test blocks: unspellchecked field',
+    },
+    {
+      kind: 'block',
+      type: 'test_unspellchecked_field',
+    },
+    {
+      kind: 'block',
+      type: 'test_parent_block',
+      inputs: {
+        INPUT: {
+          shadow: {
+            type: 'test_unspellchecked_field',
+          },
+        },
+      },
+    },
+    {
+      kind: 'block',
+      type: 'test_unspellchecked_field_and_label',
+    },
+    {
+      kind: 'block',
+      type: 'test_parent_block',
+      inputs: {
+        INPUT: {
+          shadow: {
+            type: 'test_unspellchecked_field_and_label',
+          },
+        },
+      },
+    },
+  ],
+};
+
+/**
+ * Uninstall the base multiline text block and its associated generators.
+ * TODO(#2194): remove this when those blocks are removed from the core library.
+ */
+function uninstallBlock() {
+  delete Blockly.Blocks['text_multiline'];
+  delete javascriptGenerator.forBlock['text_multiline'];
+  delete dartGenerator.forBlock['text_multiline'];
+  delete luaGenerator.forBlock['text_multiline'];
+  delete pythonGenerator.forBlock['text_multiline'];
+  delete phpGenerator.forBlock['text_multiline'];
+}
```

Comment:
Why  wrap this in a function that is called from `createWorkspace`, rather than just executing the contents directly a the top level any time before calling `document.addEventListener`?

Reply:
I find it easy to lose track of where things that are specific to the plugin test are, so I like putting it all into `createWorkspace`. This is a slight personal preference.